src/GitTools.jl

```diff
@@ -251,19 +251,18 @@ function blob_hash(path::AbstractString, HashType = SHA.SHA1_CTX)
 end
 
 """
-    contains_no_files(root::AbstractString)
+    contains_files(root::AbstractString)
 
-Helper function to determine whether a directory contains no files; e.g. it is
-either empty or contains only other directories that contain nothing but other
-directories.  This is used to exclude directories from tree hashing.
+Helper function to determine whether a directory contains files; e.g. it is a
+direct parent of a file or it contains some other directory that itself is a
+direct parent of a file. This is used to exclude directories from tree hashing.
 """
-function contains_no_files(root::AbstractString)
-    for (root, dirs, files) in walkdir(root)
-        if !isempty(files)
-            return false
-        end
+function contains_files(path::AbstractString)
+    isdir(lstat(path)) || return true
```

Comment:
```suggestion
    st = lstat(path)
    ispath(st) && !isdir(st) && return true
```
This handles non-existent paths by returning `false`, which seems like the right answer. Perhaps erroring if called on a non-existent path would be better though.